### PR TITLE
Run callback for shutdown also when container is already shut down

### PR DESCRIPTION
**Pull request by timbq**
see #1561

### CODE REVIEW

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java [hunk 1]
```diff
@@ -641,6 +641,9 @@ private void shutdownAndWaitOrCallback(@Nullable Runnable callback) {
 			}
 			else {
 				logger.info("Shutdown ignored - container is already stopped");
+				if (callback != null) {
+					getTaskExecutor().execute(callback);
```
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java:645]: There is no reason to call this callback on a different thread.

Plus, take a look, please, there is one more `return` in this method where we have to take care about this callback, too.

**Review by artembilan — CHANGES_REQUESTED** [review]
Please, consider also to add your name to the `@author` list of those changed classes.

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java (hunk 1 above)
Comment (timbq) [reply spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java:645]: You are right! I will fix this and cover the other case as soon as I get around to work on this again.

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java [hunk 2]
```diff
@@ -686,6 +687,12 @@ private void shutdownAndWaitOrCallback(@Nullable Runnable callback) {
 		}
 	}
 
+	private void runCallbackIfNotNull(@Nullable Runnable callback) {
```
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java:690]:
```suggestion
	private static void runCallbackIfNotNull(@Nullable Runnable callback) {
```

**Review by artembilan — CHANGES_REQUESTED** [review]

**Comment by artembilan** [comment]
... and cherry-picked to `2.4.x`

@timbq,

thank you for contribution; looking forward for more!

**Pull request merged**